### PR TITLE
Air pollution prediction 

### LINKED ISSUE #508: Air Pollution Prediction
The aim behind this project is to predict air pollution index. 
The code will be tested under Python and Tensorflow.
I am a participant of GSSOC'21. 
Can i please work on it?

### PR DESCRIPTION
**Pull request by Pranjal-2001**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

## Fixes #508

Replace `issue_no` with the issue number which is fixed in this PR

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [ ] Yes
- [ ] No

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

## Checklist:

- [ ] My code follows the style guidelines(Clean Code) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [ ] My changes generate no new warnings
- [ ] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

File: Air pollution prediction/training.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+gpr = GaussianProcessRegressor(normalize_y=True).fit(X_train_df,y_train)
```
Comment (antrikshmisri) [inline Air pollution prediction/training.py:1]: You are importing/defining this GaussianProcessRegressor , X_train_df , y_train anywhere

File: Air pollution prediction/plotting.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import numpy as np
+from matplotlib import pyplot as plt 
+
+
+fig,ax = plt.subplots(figsize=(12,8))
+
+plot_start = 225
+# plot the training data 
+ax.plot(y_train.index[plot_start:],y_train.values[plot_start:],'navy',marker='o',label='observed')
```
Comment (antrikshmisri) [inline Air pollution prediction/plotting.py:9]: Same thing over here and in line `11` ,`12`,`14`,`15`,`16` ,  you haven't defined Y_train , pred value anywhere

**Review by antrikshmisri — CHANGES_REQUESTED** [review]
Please see the suggestions and work on it. Also run/test your code before pushing

File: Air pollution prediction/plotting.py (hunk 2 above)
Comment (kaustubhgupta) [reply Air pollution prediction/plotting.py:9]: She has split the code into several files without any link to each other. @Pranjal-2001 You need to structure your code as done in PR #556 or #638

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by kaustubhgupta** [comment]
@Pranjal-2001 updates?

**Comment by Pranjal-2001** [comment]
I am working on it sir

**Comment by kaustubhgupta** [comment]
@Pranjal-2001 you need to make the changes today only else this issue will be assigned to someone else

**Comment by Pranjal-2001** [comment]
Could you please check the changes I made?

File: Air pollution prediction/code.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import requests
+city = 'kanpur'
```
Comment (antrikshmisri) [inline Air pollution prediction/code.py:2]:
```suggestion
city = input('Enter the city name')
```

File: Air pollution prediction/code.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+url = 'http://api.waqi.info/feed/' + city + '/?token='
+api_key = '37f96394ffe8b6cca1110af3d8270604c711c688'
```
Comment (antrikshmisri) [inline Air pollution prediction/code.py:4]: You have exposed your api key here , either replace it like this:-
```suggestion
api_key = '<Enter your api key here>' 
```
Or create a .env file and place the api key there

File: Air pollution prediction/code.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+main_url = url + api_key
+r = requests.get(main_url)
+data = r.json()['data']
+data
```
Comment (antrikshmisri) [inline Air pollution prediction/code.py:9]:
```suggestion

```

File: Air pollution prediction/code.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+print(f'{city} AQI :',aqi,'\n')
+print('Individual Air quality')
+print('Dew :',dew)
+print('no2 :',no2)
+print('Ozone :',o3)
+print('sulphur :',so2)
+print('pm10 :',so2)
+print('pm25 :',pm25)
+import matplotlib.pyplot as plt
```
Comment (antrikshmisri) [inline Air pollution prediction/code.py:32]: Remove the import statement from here and place it on top

**Review by antrikshmisri — CHANGES_REQUESTED** [review]
Refer the suggestions listed below

**Review by kaustubhgupta — COMMENTED** [review]
Wait, in the issue description, you mentioned that it is based on Python and tensorflow:
```
The aim behind this project is to predict air pollution index.
The code will be tested under Python and Tensorflow.
I am a participant of GSSOC'21.
Can i please work on it?
```

I am asking this because I assigned this issue level3 in lieu of any ML model. Forget about the ML model, this is a simple GET request to an API. @antrikshmisri @santushtisharma10 what should we do now? Decrease level of issue, suggest GUI for this request app or anything else?

**Comment by antrikshmisri** [comment]
> Wait, in the issue description, you mentioned that it is based on Python and tensorflow:
> 
> ```
> The aim behind this project is to predict air pollution index.
> The code will be tested under Python and Tensorflow.
> I am a participant of GSSOC'21.
> Can i please work on it?
> ```
> 
> I am asking this because I assigned this issue level3 in lieu of any ML model. Forget about the ML model, this is a simple GET request to an API. @antrikshmisri @santushtisharma10 what should we do now? Decrease level of issue, suggest GUI for this request app or anything else?

@kaustubhgupta Right , this doesn't implement ML in any way its a simple GET request. I think it would be appropriate to just decrease the level.

**Comment by santushtisharma10** [comment]
I agree with @antrikshmisri, we should decrease the level.

**Comment by kaustubhgupta** [comment]
> I agree with @antrikshmisri, we should decrease the level.

Level 1 ?

**Comment by santushtisharma10** [comment]
> > I agree with @antrikshmisri, we should decrease the level.
> 
> Level 1 ? As it's non gui

Yes 👍

**Comment by kaustubhgupta** [comment]
> > > I agree with @antrikshmisri, we should decrease the level.
> > 
> > 
> > Level 1 ? As it's non gui
> 
> Yes 👍

okay, done

File: Air pollution prediction/README.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+## How to use?
+
+You can clone the repository directly from 
+
+```https://github.com/Pranjal-2001/Amazing-Python-Scripts.git```
+
+## To get your API
+ 
+ 
+[Click here to get your API](http://api.waqi.info/)
```
Comment (kaustubhgupta) [inline Air pollution prediction/README.md:25]: link doesn't work

File: Air pollution prediction/README.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+_______________________________________________________________________
+
+## What is it?
+
+This helps you find the levels of Air pollution plotted graphically with a provided dataset.
+
+## TechStack
+
+1. Modules used
+       * numpy
```
Comment (kaustubhgupta) [inline Air pollution prediction/README.md:11]: There is no use of numpy, pandas, and sklearn here. You can remove it

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Code is working, just some minor changes:

**Comment by kaustubhgupta** [comment]
@Pranjal-2001 Did you look at the suggested changes?

**Comment by Pranjal-2001** [comment]
Yes sir I created a PR

**Review by kaustubhgupta — APPROVED** [review]

**Comment by kaustubhgupta** [comment]
@Kushal997-das

**Review by santushtisharma10 — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM..🚩

**Review by Kushal997-das — COMMENTED** [review]
LGTM 💯

**Pull request merged**